packages/core/src/utils/getSchemaField.ts

```diff
@@ -13,21 +14,38 @@ const getSchemaField = (
   name: string,
   schema: FormupYupSchema,
   options?: getSchemaFieldOptions,
-) => {
+): FormupYupSchema => {
   // Support nested fields
-  let fieldPath = String(name || '')
+  const originalPath = String(name || '')
     .split('.')
     .reduce((prev, curr) => `${prev}.fields.${curr}`);
 
-  // Remove array indexes
-  if (fieldPath.includes('[')) {
+  let fieldPath = originalPath;
+
+  // Get array schema
+  if (originalPath.includes('[')) {
     fieldPath = fieldPath.substring(0, fieldPath.indexOf('['));
   }
 
   const schemaField = get(schema?.fields || {}, fieldPath);
 
   if (options?.returnSubtype) {
-    return schemaField?.[yupSchemaFieldProperties.subType];
+    if (schemaField?.[yupSchemaFieldProperties.type] === fieldTypes.array) {
+      const nestedPath = originalPath.replace(/\.fields/g, '').split('.');
+      const subType = schemaField?.[yupSchemaFieldProperties.subType];
+      const isSchemaObject = subType?.[yupSchemaFieldProperties.type] === fieldTypes.object;
```

Comment:
Maybe improve on the name here?

```suggestion
      const isObjectType = subType?.[yupSchemaFieldProperties.type] === fieldTypes.object;
```

Reply:
Yes, looks better indeed.